pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -298,63 +350,75 @@ func (e *DockercfgController) syncServiceAccount(key string) error {
 	return err
 }
 
-const (
-	tokenSecretWaitInterval = 20 * time.Millisecond
-	tokenSecretWaitTimes    = 100
-)
+// getTokenSecret return a token secret for a given service account created by this
+// controller or returns nil when there is no token secret created yet.
+func (e *DockercfgController) getTokenSecret(serviceAccount *api.ServiceAccount) *api.Secret {
+	var tokenSecret *api.Secret
+	for _, s := range e.secretsCache.ListKeys() {
```

Comment:
this should only pay attention to secrets this controller created (you should annotate them on creation so you can identity them)